src/main/scala/Cqrs/Aggregate.scala

```diff
@@ -23,9 +28,9 @@ object Aggregate {
   final case class VersionedEvents[E](version: Int, events: List[E])
 
   sealed trait EventDatabaseOp[E, A]
-  final case class ReadAggregateExistence[E](id: AggregateId) extends EventDatabaseOp[E, Error Xor Boolean]
-  final case class ReadAggregate[E](id: AggregateId, fromVersion: Int) extends EventDatabaseOp[E, Error Xor List[VersionedEvents[E]]]
-  final case class AppendAggregateEvents[E](id: AggregateId, events: VersionedEvents[E]) extends EventDatabaseOp[E, Error Xor Unit]
+  final case class ReadAggregateExistence[E](tag: Tag, id: AggregateId) extends EventDatabaseOp[E, Error Xor Boolean]
```

Comment:
I see, makes abit sense. However, how would you read all events of a counter for example without knowing all the aggregateIds? `case class ReadAggregate[E](tag: Tag, id: AggregateId, fromVersion: Int)` requires you to specify a AggregateId